### ISSUE TITLE
[iOS] Frame doesn't show contents if I use native styles

### ISSUE THREAD

**Issue by Daoting** [unoplatform/uno#8190]
### Current behavior

Frame doesn't show contents after call `FeatureConfiguration.Style.ConfigureNativeFrameNavigation()`.

### How to reproduce it (as minimally and precisely as possible)

Simply adding this line to the App.xaml.cs constructor after the `InitializeComponent()` call should do the trick
`FeatureConfiguration.Style.ConfigureNativeFrameNavigation()`

### Workaround

The log contents:
```
2022-02-25 08:24:36.447983+0800 UnoApp.Mobile[63947:393024] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=63947
2022-02-25 08:24:36.448388+0800 UnoApp.Mobile[63947:393024] SecTaskCopyDebugDescription: UnoApp.Mobile[63947]/0#-1 LF=0
2022-02-25 08:24:36.458500+0800 UnoApp.Mobile[63947:393024] SecTaskLoadEntitlements failed error=22 cs_flags=200, pid=63947
2022-02-25 08:24:36.458988+0800 UnoApp.Mobile[63947:393024] SecTaskCopyDebugDescription: UnoApp.Mobile[63947]/0#-1 LF=0
Thread started: <Thread Pool> #2
Thread started: .NET ThreadPool Gate #3
Thread started: <Thread Pool> #4
Thread started: <Thread Pool> #5
Thread started: <Thread Pool> #6
Thread started: .NET Timers #7
Thread started: .NET Sockets #8
2022-02-25 08:24:38.132130+0800 UnoApp.Mobile[63947:393024] fail: Microsoft.UI.Xaml.BrushTransition[0] The member BrushTransition.BrushTransition() is not implemented in Uno.
2022-02-25 08:24:38.132720+0800 UnoApp.Mobile[63947:393024] fail: Microsoft.UI.Xaml.BrushTransition[0] The member TimeSpan BrushTransition.Duration is not implemented in Uno.
2022-02-25 08:24:38.133009+0800 UnoApp.Mobile[63947:393024] fail: Microsoft.UI.Xaml.Controls.ContentPresenter[0] The member BrushTransition ContentPresenter.BackgroundTransition is not implemented in Uno.
2022-02-25 08:24:38.287938+0800 UnoApp.Mobile[63947:393024] fail: Microsoft.UI.Xaml.FrameworkElement[0] Failed to process MoveToWindow for Microsoft.UI.Xaml.Controls.Frame
ObjCRuntime.RuntimeException: Failed to lookup the required marshalling information.
Additional information:
	Selector: setFrame:
	Type: Uno_UI_Controls_UnoNavigationBar

 ---> ObjCRuntime.RuntimeException: Failed to get the 'this' instance in a method call to Uno.UI.Controls.UnoNavigationBar.set_Frame.
 ---> ObjCRuntime.RuntimeException: Failed to marshal the Objective-C object 0x7fe29395b480 (type: Uno_UI_Controls_UnoNavigationBar). Could not find an existing managed instance for this object, nor was it possible to create a new managed instance (because the type 'Uno.UI.Controls.UnoNavigationBar' does not have a constructor that takes one NativeHandle argument).
   at ObjCRuntime.Runtime.MissingCtor(IntPtr ptr, IntPtr klass, Type type, MissingCtorResolution resolution)
   at ObjCRuntime.Runtime.ConstructNSObject[NSObject](IntPtr ptr, Type
 type, MissingCtorResolution missingCtorResolution)
   at ObjCRuntime.Runtime.ConstructNSObject(IntPtr ptr, IntPtr klass, MissingCtorResolution missingCtorResolution)
   at ObjCRuntime.Runtime.GetNSObject(IntPtr ptr, MissingCtorResolution missingCtorResolution, Boolean evenInFinalizerQueue)
   at Registrar.DynamicRegistrar.GetMethodDescriptionAndObject(Type type, IntPtr selector, Boolean is_static, IntPtr obj, IntPtr& mthis, IntPtr desc)
   --- End of inner exception stack trace ---
   at Registrar.DynamicRegistrar.GetMethodDescriptionAndObject(Type type, IntPtr selector, Boolean is_static, IntPtr obj, IntPtr& mthis, IntPtr desc)
   at ObjCRuntime.Runtime.GetMethodAndObjectForSelector(IntPtr klass, IntPtr sel, Boolean is_static, IntPtr obj, IntPtr& mthis, IntPtr desc)
   at ObjCRuntime.Runtime.get_method_and_object_for_selector(IntPtr cls, IntPtr sel, Boolean is_static, IntPtr obj, IntPtr& mthis, IntPtr desc, IntPtr& exception_gchandle)
   --- End of inner exception stack trace ---
   at UIKit.UINavigationCon
troller.set_NavigationBarHidden(Boolean value)
   at Uno.UI.Controls.NativeFramePresenter..ctor()
   at Uno.UI.__Resources._Generic_Native_c3cd4f00dc193686b37435bd25614734_Generic_NativeRDSC9.Build(Object __ResourceOwner_1963)
   at Uno.UI.GlobalStaticResources.ResourceDictionarySingleton__0.<>c.<Get_0_13>b__18_1(Object __owner)
   at Microsoft.UI.Xaml.FrameworkTemplate.LoadContent()
   at Microsoft.UI.Xaml.FrameworkTemplatePool.DequeueTemplate(FrameworkTemplate template)
   at Microsoft.UI.Xaml.FrameworkTemplate.LoadContentCached()
   at Microsoft.UI.Xaml.Controls.Control.UpdateTemplate()
   at Microsoft.UI.Xaml.Controls.Control.SetUpdateControlTemplate(Boolean forceUpdate)
   at Microsoft.UI.Xaml.Controls.Control.OnTemplateChanged(DependencyPropertyChangedEventArgs e)
   at Microsoft.UI.Xaml.Controls.Control.<>c.<.cctor>b__382_0(DependencyObject s, DependencyPropertyChangedEventArgs e)
   at Microsoft.UI.Xaml.PropertyMetadata.RaisePropertyChanged(DependencyObject source, DependencyPropertyChangedEventArgs e
)
   at Microsoft.UI.Xaml.DependencyObjectStore.InvokeCallbacks(DependencyObject actualInstanceAlias, DependencyProperty property, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence, Boolean bypassesPropagation)
   at Microsoft.UI.Xaml.DependencyObjectStore.RaiseCallbacks(DependencyObject actualInstanceAlias, DependencyPropertyDetails propertyDetails, Object previousValue, DependencyPropertyValuePrecedences previousPrecedence, Object newValue, DependencyPropertyValuePrecedences newPrecedence)
   at Microsoft.UI.Xaml.DependencyObjectStore.InnerSetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails propertyDetails, Boolean isPersistentResourceBinding)
   at Microsoft.UI.Xaml.DependencyObjectStore.SetValue(DependencyProperty property, Object value, DependencyPropertyValuePrecedences precedence, DependencyProper
tyDetails propertyDetails, Boolean isPersistentResourceBinding)
   at Microsoft.UI.Xaml.DependencyObjectStore.SetValue(DependencyProperty property, Object value)
   at Uno.UI.Controls.BindableUIView.SetValue(DependencyProperty dp, Object value)
   at Microsoft.UI.Xaml.Setter.ApplyTo(DependencyObject o)
   at Microsoft.UI.Xaml.Style.ApplyTo(DependencyObject o, DependencyPropertyValuePrecedences precedence)
   at Microsoft.UI.Xaml.FrameworkElement.OnStyleChanged(Style oldStyle, Style newStyle, DependencyPropertyValuePrecedences precedence)
   at Microsoft.UI.Xaml.FrameworkElement.ApplyDefaultStyle()
   at Microsoft.UI.Xaml.FrameworkElement.ApplyStyles()
   at Microsoft.UI.Xaml.FrameworkElement.OnLoadingPartial()
   at Microsoft.UI.Xaml.FrameworkElement.OnLoading()
   at Microsoft.UI.Xaml.FrameworkElement.MovedToWindow()
```

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.2.100

### Affected platforms

iOS

### IDE

Visual Studio 2022

### IDE version

17.2 preview1

**Comment by jeromelaban** [comment]
Thanks, this is a net6 issue which is fixed in Uno 4.2, given that [.NET 6 Preview 13 is installed](https://github.com/unoplatform/uno/discussions/7833).

**Comment by Daoting** [comment]
Thanks for quick response, @jeromelaban!
